azkaban-common/src/main/java/azkaban/project/AzkabanProjectLoader.java

```diff
@@ -80,8 +80,46 @@
       throws ProjectManagerException {
     log.info("Uploading files to " + project.getName());
 
-    // Unzip.
-    File file = null;
+    // Since props is an instance variable of ProjectManager, and each
+    // invocation to the uploadProject manager needs to pass a different
+    // value for the PROJECT_ARCHIVE_FILE_PATH key, it is necessary to
+    // create a new instance of Props to make sure these different values
+    // are isolated from each other.
+    final Props prop = new Props(this.props);
```

Comment:
new Props(this.props.getParent(), this.props) ?

Reply:
Here prop doesn't have to be exactly the same. It can just inherit the parent prop. I think either way is fine. I didn't change the original implementation here, just moved it to a new method.